File: sdkwork-app-sdk-go/api/app_documents.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/app-sdk/types"
    sdkhttp "github.com/sdkwork/app-sdk/http"
)

type AppDocumentsApi struct {
    client *sdkhttp.Client
}

func NewAppDocumentsApi(client *sdkhttp.Client) *AppDocumentsApi {
    return &AppDocumentsApi{client: client}
}

// Get document detail
func (a *AppDocumentsApi) GetDocumentDetail(documentId string) (sdktypes.PlusApiResultDocumentDetailVO, error) {
    raw, err := a.client.Get(AppApiPath(fmt.Sprintf("/documents/%s", documentId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentDetailVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentDetailVO](raw)
}

// Update document metadata
func (a *AppDocumentsApi) UpdateDocument(documentId string, body sdktypes.DocumentUpdateRequest) (sdktypes.PlusApiResultDocumentOperationVO, error) {
    raw, err := a.client.Put(AppApiPath(fmt.Sprintf("/documents/%s", documentId)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentOperationVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentOperationVO](raw)
}

// Delete document
func (a *AppDocumentsApi) DeleteDocument(documentId string) (sdktypes.PlusApiResultVoid, error) {
    raw, err := a.client.Delete(AppApiPath(fmt.Sprintf("/documents/%s", documentId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultVoid
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVoid](raw)
}

// Get document content
func (a *AppDocumentsApi) GetDocumentContent(documentId string) (sdktypes.PlusApiResultDocumentContentVO, error) {
    raw, err := a.client.Get(AppApiPath(fmt.Sprintf("/documents/%s/content", documentId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentContentVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentContentVO](raw)
}

// Update document content
func (a *AppDocumentsApi) UpdateDocumentContent(documentId string, body sdktypes.DocumentContentUpdateRequest) (sdktypes.PlusApiResultDocumentContentVO, error) {
    raw, err := a.client.Put(AppApiPath(fmt.Sprintf("/documents/%s/content", documentId)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentContentVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentContentVO](raw)
}

// List documents
func (a *AppDocumentsApi) ListDocuments(query map[string]interface{}) (sdktypes.PlusApiResultPageDocumentVO, error) {
    raw, err := a.client.Get(AppApiPath("/documents"), query, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultPageDocumentVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultPageDocumentVO](raw)
}

// Create document
func (a *AppDocumentsApi) CreateDocument(body sdktypes.DocumentCreateRequest) (sdktypes.PlusApiResultDocumentOperationVO, error) {
    raw, err := a.client.Post(AppApiPath("/documents"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentOperationVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentOperationVO](raw)
}

// Copy document
func (a *AppDocumentsApi) CopyDocument(documentId string, body sdktypes.DocumentCopyRequest) (sdktypes.PlusApiResultDocumentOperationVO, error) {
    raw, err := a.client.Post(AppApiPath(fmt.Sprintf("/documents/%s/copy", documentId)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentOperationVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentOperationVO](raw)
}

// Batch update document
func (a *AppDocumentsApi) BatchUpdateDocument(documentId string, body sdktypes.DocumentBatchUpdateRequest) (sdktypes.PlusApiResultDocumentBatchUpdateResultVO, error) {
    raw, err := a.client.Post(AppApiPath(fmt.Sprintf("/documents/%s/batch-update", documentId)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentBatchUpdateResultVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentBatchUpdateResultVO](raw)
}

// Batch update document
func (a *AppDocumentsApi) CreateBatchUpdateDocument(documentId string, body sdktypes.DocumentBatchUpdateRequest) (sdktypes.PlusApiResultDocumentBatchUpdateResultVO, error) {
    raw, err := a.client.Post(AppApiPath(fmt.Sprintf("/documents/%s:batchUpdate", documentId)), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultDocumentBatchUpdateResultVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultDocumentBatchUpdateResultVO](raw)
}

File: sdkwork-app-sdk-typescript/src/api/app-documents.ts
import { appApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { DocumentBatchUpdateRequest, DocumentContentUpdateRequest, DocumentCopyRequest, DocumentCreateRequest, DocumentUpdateRequest, PlusApiResultDocumentBatchUpdateResultVO, PlusApiResultDocumentContentVO, PlusApiResultDocumentDetailVO, PlusApiResultDocumentOperationVO, PlusApiResultPageDocumentVO, PlusApiResultVoid } from '../types';


export class AppDocumentsApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Get document detail */
  async getDocumentDetail(documentId: string | number): Promise<PlusApiResultDocumentDetailVO> {
    return this.client.get<PlusApiResultDocumentDetailVO>(appApiPath(`/documents/${documentId}`));
  }

/** Update document metadata */
  async updateDocument(documentId: string | number, body: DocumentUpdateRequest): Promise<PlusApiResultDocumentOperationVO> {
    return this.client.put<PlusApiResultDocumentOperationVO>(appApiPath(`/documents/${documentId}`), body);
  }

/** Delete document */
  async deleteDocument(documentId: string | number): Promise<PlusApiResultVoid> {
    return this.client.delete<PlusApiResultVoid>(appApiPath(`/documents/${documentId}`));
  }

/** Get document content */
  async getDocumentContent(documentId: string | number): Promise<PlusApiResultDocumentContentVO> {
    return this.client.get<PlusApiResultDocumentContentVO>(appApiPath(`/documents/${documentId}/content`));
  }

/** Update document content */
  async updateDocumentContent(documentId: string | number, body: DocumentContentUpdateRequest): Promise<PlusApiResultDocumentContentVO> {
    return this.client.put<PlusApiResultDocumentContentVO>(appApiPath(`/documents/${documentId}/content`), body);
  }

/** List documents */
  async listDocuments(params?: QueryParams): Promise<PlusApiResultPageDocumentVO> {
    return this.client.get<PlusApiResultPageDocumentVO>(appApiPath(`/documents`), params);
  }

/** Create document */
  async createDocument(body: DocumentCreateRequest): Promise<PlusApiResultDocumentOperationVO> {
    return this.client.post<PlusApiResultDocumentOperationVO>(appApiPath(`/documents`), body);
  }

/** Copy document */
  async copyDocument(documentId: string | number, body: DocumentCopyRequest): Promise<PlusApiResultDocumentOperationVO> {
    return this.client.post<PlusApiResultDocumentOperationVO>(appApiPath(`/documents/${documentId}/copy`), body);
  }

/** Batch update document */
  async batchUpdateDocument(documentId: string | number, body: DocumentBatchUpdateRequest): Promise<PlusApiResultDocumentBatchUpdateResultVO> {
    return this.client.post<PlusApiResultDocumentBatchUpdateResultVO>(appApiPath(`/documents/${documentId}/batch-update`), body);
  }

/** Batch update document */
  async createBatchUpdateDocument(documentId: string | number, body: DocumentBatchUpdateRequest): Promise<PlusApiResultDocumentBatchUpdateResultVO> {
    return this.client.post<PlusApiResultDocumentBatchUpdateResultVO>(appApiPath(`/documents/${documentId}:batchUpdate`), body);
  }
}

export function createAppDocumentsApi(client: HttpClient): AppDocumentsApi {
  return new AppDocumentsApi(client);
}
